File: Programowanie obiektowe/Lista5/Zad2/Odejmij.java
package com.company;

/**
 * Created by teez on 25.03.2016.
 */
public class Odejmij extends wyrazenie {

    int a;
    int b;


    public Odejmij(wyrazenie a, wyrazenie b){
        this.a = a.oblicz();
        this.b = b.oblicz();
    }
    public String toString(){
        return " Wynikiem bedzie suma " + this.a + " - " +  this.b;
    }

    public int oblicz() {
        return this.a-this.b;
    }
}

File: Kurs C++/Zad6/odejmij.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "odejmij.h"


double odejmij::oblicz() {
    return a->oblicz()-b->oblicz();
}
odejmij::odejmij(wyrazenie *a, wyrazenie *b) { this->a = a; this->b = b; }


std::string odejmij::formatuj_wyrazenie() {
    return "(" + a->formatuj_wyrazenie() + " - " + b->formatuj_wyrazenie() + ")";
}